### PR TITLE
fix GoCardless institutions with special continuous access EUA behaviour

### PR DESCRIPTION
**Pull request by matt-fidd**
Fixes bug reported in https://github.com/actualbudget/actual/pull/5714#issuecomment-3419530826

GoCardless publish a list of banks that behave like this (https://bankaccountdata.zendesk.com/hc/en-gb/articles/11529718632476-Extended-history-and-continuous-access-edge-cases).

### CODE REVIEW

**Comment by matt-fidd** [comment]
@michele-perrone are you able to test this change please?

File: packages/sync-server/src/app-gocardless/bank-factory.js [hunk 1]
```diff
@@ -29,3 +29,171 @@
...
+  'POSTE_PAY_BPPIITRRXXX',
+  'PRADER_BANK_PRRDIT21XXX',
+  'ROMAGNABANCA_CCRTIT2TRMB',
+  'ROVIGO_BANCA_ICRAITRRCK0',
+  'SARDEGNA_CORP_SARDIT31',
+  'SARDEGNA_RETAIL_SARDIT31',
+  'SOLUTION_BANK_CRDRIT2FXXX',
+  'WIDIBA_WIDIITMM',
+  'ZADRUZNA_KRASKA_BANKA_TRST_GORICA_ZADRUGA_CCRTIT2TV00',
+].map(i => i.replace('*', ''));
```
[the remark this answers is not in the document]
Comment (matt-fidd) [reply packages/sync-server/src/app-gocardless/bank-factory.js:190]: It wouldn't work if there was more than one anyway, the function that uses this uses startsWith to check so if there was a wildcard in the middle it wouldn't match.

**Comment by michele-perrone** [comment]
Hello @matt-fidd , thanks a lot for the PR!

I re-added the bank sync, and I got my new transactions (this was expected).
Now I have to see if new GoCardless requests will pull new transactions or not :-)
I'll keep you posted! I'll check back in a few days.

**Comment by michele-perrone** [comment]
Hello Matt, I can confirm that this works! Today I launched the bank sync and I got new transactions.
Thanks again for the effort! This will fix a lot of banks.

**Comment by amenrio** [comment]
Hello! I've also had issues with this before in #5709
You mention @matt-fidd 
> GoCardless publish a list of banks that behave like this (https://bankaccountdata.zendesk.com/hc/en-gb/articles/11529718632476-Extended-history-and-continuous-access-edge-cases).

I'm surprised my bank (`ABANCA_CAGLESMM`) is not listed in the GoCardless list.

I'm curious to test this changes as well but I'm not sure how would I do that, Do i need to build the app from source?
Would be ok to export the data of my instance, pop up a new build from source, grab that export backup and run it on the newer build? 

Thanks in advance
